### PR TITLE
EA-2832: Use regexes for validating URIs

### PR DESCRIPTION
**Pull request by ikattey**
* only http / https schemes supported in URLs

### CODE REVIEW

File: entity-management-definitions/src/main/java/eu/europeana/entitymanagement/utils/UriValidator.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+  private static final Predicate<String> URN_MATCH_PREDICATE =
+      Pattern.compile(
+              "^urn:[a-z0-9][a-z0-9-]{0,31}:([a-z0-9()+,\\-.:=@;$_!*']|%[0-9a-f]{2})+$",
+              Pattern.CASE_INSENSITIVE)
+          .asMatchPredicate();
+
+  // only http / https schemes supported, non-latin characters should be percent-encoded
+  private static final Predicate<String> URL_MATCH_PREDICATE =
+      Pattern.compile("^(https?)://[-a-zA-Z0-9+&@#/%?=~_|!:,.;]*[-a-zA-Z0-9+&@#/%=~_|]")
```
Comment (ikattey) [inline entity-management-definitions/src/main/java/eu/europeana/entitymanagement/utils/UriValidator.java]: To support additional schemes (eg ftp / ftps), `(https?)` can be changed to `(https?|ftps?)`.

File: entity-management-definitions/src/main/java/eu/europeana/entitymanagement/normalization/EntityFieldsCompleteValidator.java [hunk 2]
```diff
@@ -120,21 +115,14 @@ boolean validateStringValue(
...
-    } else {
-      EntityFieldsTypes fieldType = EntityFieldsTypes.valueOf(field.getName());
-      boolean isUri = isUri(fieldValue);
-      if (isUri) {
-        // allow URIs for field type TEXT OR URI but only for empty languge codes
-        if ("".equals(key)
-            && EntityFieldsTypes.FIELD_TYPE_TEXT_OR_URI.equals(fieldType.getFieldType())) {
-          return validateIRIFormat(context, field, fieldValue);
-        } else {
-          addUriNotAllowedConstraint(context, field, fieldValue, key);
```
Comment (gsergiu) [inline entity-management-definitions/src/main/java/eu/europeana/entitymanagement/normalization/EntityFieldsCompleteValidator.java:132]: this check is still needed

File: entity-management-definitions/src/main/java/eu/europeana/entitymanagement/normalization/EntityFieldsCompleteValidator.java [hunk 3]
```diff
@@ -120,21 +115,14 @@ boolean validateStringValue(
...
-        }
-      }
-      return true;
     }
+
+    EntityFieldsTypes fieldType = EntityFieldsTypes.valueOf(field.getName());
+    if (isUriField(key, fieldType.getFieldType()) && !UriValidator.isUri(fieldValue)) {
+      addUriNotAllowedConstraint(context, field, fieldValue, key);
+      return false;
+    }
```
Comment (gsergiu) [inline entity-management-definitions/src/main/java/eu/europeana/entitymanagement/normalization/EntityFieldsCompleteValidator.java:133]: if not isUriField, we still have to validate that the value is not an URI according to the specs - Text Must be a string (not a reference) see also previous comment

**Review by gsergiu — COMMENTED** [review]
the validation of string values (DataType Text) still needs to check if the value is not URI

**Pull request merged**